addons/web/models/models.py

```diff
@@ -146,7 +146,7 @@ def search_panel_select_range(self, field_name):
             fields.append(parent_name)
         return {
             'parent_field': parent_name,
-            'values': Comodel.search_read([], fields),
+            'values': Comodel.with_context({'in_search_panel': True}).search_read([], fields),
```

Comment:
This is horribly wrong: calling `with_context` with a dict **throws away** the whole context!
Call it with a keyword argument instead:
```suggestion
            'values': Comodel.with_context(in_search_panel=True).search_read([], fields),
```

I also suggest to use a key less specific to the search panel, like `hierarchical_naming=False`, or `small_name=True`, or `long_name=False`...